### ISSUE TITLE
Template folder should use the file id

### ISSUE THREAD

**Issue by juliushaertl** [nextcloud/server#31414]
As per comment in https://github.com/nextcloud/server/pull/28645#issuecomment-1056526796 the template folder should be stored using the file id of the folder instead of the path of a user which has the benefit that it would survive the rename or move of the folder itself.

I'd suggest to do a rolling upgrade of that part so that if we detect a path in the config, we switch it to the file id directly and continue to use that further on.

**Comment by Jerome-Herbinet** [comment]
I totally agree with you @juliushaertl. 

I hesitated to create a new "issue" with all that I have to explain, but I'll try like this, in your issue.

I would add that other things are really missing to make the ergonomics good (Can you give me your opinion on each of these points?) :

1. The folder, once chosen or created, must have the following icon so that it can be differentiated from the others (in the web interface as well as in the mobile applications). Currently, the folder has a normal folder icon, which can be annoying; it can be deleted by mistake, or difficult to find.
![2022-07-21_11-14](https://user-images.githubusercontent.com/33763786/180177441-23d78a51-ba96-4aac-99be-6a7a42ed1c06.png)

2. The "set up templates folder" option should not disappear after you have defined it ... because you may need to define another folder one day instead of the one that was initially chosen. Currently, this disappearance and inability to redefine it may be seen as a bug. There is a trick I discovered, which consists in renaming, moving or deleting the folder currently chosen so that the "set up templates folder" option reappears after refreshing the page ... but this is actually a bug ... this bug which is the subject of this "issue" and which is useful in this case.

3. The UX only allows you to easily create a new template folder at the root. But, in the following other cases, it is not intuitive (and most users might consider that it is not even possible, thus very limited): 
- Designating an existing folder as a template storage folder; to do this, you have to enter the exact name of an existing folder, but this is not ergonomic. Few people will have the idea of doing this, although I'm sure it must happen regularly. Some users will be clever enough to try it this way, but others will think it's not possible or be afraid of breaking something.
- Creating (or naming if it already exists) a template storage folder in an existing folder. This can be done by typing "{name of parent folder}/{name of my templates folder}". This is already not very intuitive, but you have to pay attention to two things: For this to work, you must not put a "/" before the name of the parent folder, and you must be in the root of Nextcloud Files.

4. This may be more about Collabora (RichDocuments), but I don't understand why you can't open a template with Collabora Online / Nextcloud Office. When you click on it, the file downloads but does not open in Collabora Online / Nextcloud Office. I don't see any reason why this is not possible, especially since Collabora Online / Nextcloud Office allows you to convert a normal file into a template file... and the two files are intrinsically almost identical. This is not practical for building templates in the following two cases: 
- if you want to use templates in parallel with LibreOffice (template folder synchronised on the computer with the synchronisation client)
- if the template folder is defined via the Collabora Online settings instead of Nextcloud Files
It's a good thing that when the templates folder is defined with Nextcloud Files that the normal files (odt, ods, odp ...) are considered as templates because otherwise, it would be really complicated ... or else, you have to have the odt, ods, odp, odg files in another folder and do "save as" as soon as the templates are updated and then come and overwrite them in the templates folder ... it's really not practical. This fourth point might require the opening of a separate "issue".

=============================================================================

In terms of ergonomics, instead of the current way of working, there should be (easy to say, but maybe not to do, but it would change everything), 2 possibilities : 
1. when clicking on the 3 dots (meat balls) of a folder, have an option "Set as template folder". After that, for that folder: The folder icon changes (see what I talked about in point number 1) and the "Set as template folder" option disappears.
2. When you click on "+" and then on "set up templates folder", instead of the current input field, a popup should open; inside, you can navigate through the folders and create folders with a "+" button near the breadcrumb trail, and a "choose" button. When you click on "choose", the popup disappears, the folder is defined and appears with an appropriate icon (see point number 1)

**Comment by Jerome-Herbinet** [comment]
I've just added 2 screenshots for the 2 UX/UI suggestions.

**Comment by Jerome-Herbinet** [comment]
I also noticed (to be tested again), that it is possible to designate the root folder as the template storage folder. Once this is done, since you can't rename, or move the root (which is quite understandable), you can't define a folder at all (unless you fix the problem in a parameter file or in a database ... I don't know where this is managed).

**Comment by Jerome-Herbinet** [comment]
I've moved my suggestions in a new issue (#33372).
